src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectChangeEventArgs.cs

```diff
@@ -7,29 +7,35 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
 {
     internal class ProjectChangeEventArgs : EventArgs
     {
-        public ProjectChangeEventArgs(string projectFilePath, ProjectChangeKind kind)
+        public ProjectChangeEventArgs(ProjectSnapshot older, ProjectSnapshot newer, ProjectChangeKind kind)
         {
-            if (projectFilePath == null)
+            if (older == null && newer == null)
             {
-                throw new ArgumentNullException(nameof(projectFilePath));
+                throw new ArgumentException("Both projects cannot be null.");
```

Comment:
Resource eventually, #importantfeedback

Reply:
I don't think this will ever matter. I've mostly stopped using resources. The other room doesn't use them (and mostly never did).